src/test/java/com/conveyal/datatools/manager/jobs/ArbitraryTransformJobTest.java

```diff
@@ -0,0 +1,271 @@
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import static com.conveyal.datatools.TestUtils.appendDate;
+import static com.conveyal.datatools.TestUtils.assertThatSqlCountQueryYieldsExpectedCount;
+import static com.conveyal.datatools.TestUtils.createFeedVersion;
+import static com.conveyal.datatools.TestUtils.zipFolderFiles;
+import static com.conveyal.datatools.manager.models.FeedRetrievalMethod.MANUALLY_UPLOADED;
+import static com.conveyal.datatools.manager.models.FeedRetrievalMethod.VERSION_CLONE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertThat;
+
+public class ArbitraryTransformJobTest extends UnitTest {
+    private static final Logger LOG = LoggerFactory.getLogger(ArbitraryTransformJob.class);
+    private static final Auth0UserProfile user = Auth0UserProfile.createTestAdminUser();
+    private static Project project;
+    private static FeedSource feedSource;
+    private FeedVersion sourceVersion;
+    private FeedVersion targetVersion;
+
+    /**
+     * Initialize Data Tools and set up a simple feed source and project.
+     */
+    @BeforeClass
+    public static void setUp() throws IOException {
+        // start server if it isn't already running
+        DatatoolsTest.setUp();
+
+        // Create a project, feed sources, and feed versions to merge.
+        project = new Project();
+        project.name = appendDate("Test");
+        Persistence.projects.create(project);
+
+        // Bart
+        feedSource = new FeedSource(appendDate("Test Feed"), project.id, MANUALLY_UPLOADED);
+        Persistence.feedSources.create(feedSource);
+    }
+
+    /**
+     * Clean up test database after tests finish.
+     */
+    @AfterClass
+    public static void tearDown() {
+        // Project delete cascades to feed sources.
+        project.delete();
+    }
+
+    /**
+     * Run set up before each test. This just resets the feed source transformation properties.
+     */
+    @Before
+    public void setUpTest() {
+        feedSource = Persistence.feedSources.getById(feedSource.id);
+        feedSource.transformRules = new ArrayList<>();
+        Persistence.feedSources.replace(feedSource.id, feedSource);
+    }
+
+    /**
+     * Run tear down after each test. This just deletes the feed versions that were used in the test.
+     */
+    @After
+    public void tearDownTest() {
+        // Clean up
+        if (sourceVersion != null) sourceVersion.delete();
+        if (targetVersion != null) targetVersion.delete();
+    }
+
+    /**
+     * Test that a {@link ReplaceFileFromVersionTransformation} will successfully add a GTFS+ file found in the source version
+     * into the target version's GTFS file.
+     */
+    @Test
+    public void canReplaceGtfsPlusFileFromVersion() throws IOException {
```

Comment:
On Windows you get the following warning which in turn fails this test:

java.nio.file.FileSystemException: C:\tmp\Test_Feed_Tue_Jun_30_145808_BST_2020-20200630T145809-01-619a856f-4b4d-4abb-a267-8eb353141241-1af5e4ae-c948-4320-8843-85c6beeaf5de.zip: The process cannot access the file because it is being used by another process.

This is for reference. Perhaps if I move on to working on datatools I'll attempt to resolve.